synthea_omop_etl/models/base/synthea_patients.sql

```diff
@@ -1,11 +1,27 @@
+-- Returns a list of the columns from a relation, so you can then iterate in a for loop
+{% set column_names = 
+    dbt_utils.get_filtered_columns_in_relation( source('synthea', 'patients') ) 
+%}
+
+
 WITH patients AS (
 
-    SELECT
-        "Id" AS patient_id,
-        "BIRTHDATE" AS patient_birth_date
-    FROM {{ ref('patients') }}
+    SELECT  
+
+        {% for column_name in column_names %}
+            {% if column_name == "Id" %}
+            "{{ column_name }}" as patient_id
+            {% elif column_name ==  "BIRTHDATE" %}
+            "{{ column_name }}" as  patient_birth_date
+            {% else %}
+            "{{ column_name }}" as {{ column_name | lower }}
+            {% endif %}
+        {% if not loop.last %},{% endif %}
+        {% endfor %}
+        
+    FROM {{ source('synthea','patients' ) }}
+) 
```

Comment:
to get the most out of the lowercasing function, i'd recommend to have one CTE that does the for loop and lowercases everything, _then_ another CTE where columns can be renamed as needed.